voter-dapp/src/RetrieveRewards.js

```diff
@@ -0,0 +1,201 @@
+    // should query (since only one can be chosen).
+    let oldestUnclaimedRound = MAX_SAFE_INT;
+    for (const event of revealedVoteEvents) {
+      const voteState = getVoteState(event.returnValues.identifier, event.returnValues.time);
+      const revealRound = event.returnValues.roundId.toString();
+
+      if (!voteState.retrievedRewards && voteState.priceResolutionRound === revealRound) {
+        // Reveal happened in the same round as the price resolution: definitely a retrievable reward.
+        oldestUnclaimedRound = Math.min(oldestUnclaimedRound, revealRound);
+        voteState.didReveal = true;
+      }
+    }
+
+    if (oldestUnclaimedRound === MAX_SAFE_INT) {
+      // No unclaimed rounds were found.
+      return { ready: true, status };
+    }
+
+    // Extract identifiers and times from the round we picked.
+    const toRetrieve = [];
+    const maxBatchRetrievals = 100;
+    for (const [key, voteState] of Object.entries(state)) {
+      if (
+        !voteState.retrievedRewards &&
+        voteState.priceResolutionRound === oldestUnclaimedRound.toString() &&
+        voteState.didReveal
+      ) {
+        // If this is an eligible reward for the oldest round, extract the information and push it into the retrieval array.
+        const [identifier, time] = key.split("|");
+        toRetrieve.push({ identifier: web3.utils.utf8ToHex(identifier), time: time });
+      }
+
+      // Only so many reward claims can fit in a single transaction, so break if we go over that limit.
+      if (toRetrieve.length === maxBatchRetrievals) {
+        break;
+      }
+    }
+
+    // Create the txn send function and return it.
+    const retrieveRewards = () => {
+      send(votingAccount, oldestUnclaimedRound.toString(), toRetrieve);
+    };
+
+    return { ready: true, send: retrieveRewards, status };
+  }
+}
+
+function RetrieveRewards({ votingAccount }) {
+  const { useCacheCall, useCacheEvents } = drizzleReactHooks.useDrizzle();
+  const classes = useTableStyles();
+
+  const currentRoundId = useCacheCall("Voting", "getCurrentRoundId");
+
+  // This variable tracks whether the user only wants to query a limited lookback or all history for unclaimed rewards.
+  const [queryAllRounds, setQueryAllRounds] = useState(false);
+
+  // Determines the list of roundIds to query for. Will return undefined if the user wants to search the entire history.
+  const roundIds = useMemo(() => {
+    if (queryAllRounds) {
+      // Signal that all rounds should be searched by setting the query parameter to undefined.
+      return undefined;
+    } else if (currentRoundId === undefined) {
+      // Should always return nothing, which is desired before we know the round id.
+      return MAX_UINT_VAL;
+    } else {
+      // This section will produce an array of roundIds that should be queried for unclaimed rewards.
+      const defaultLookback = 10; // Default lookback is 10 rounds.
```

Comment:
Yeah, it amounts to about 20 days. I think the big shift is that this number is now going to be in time space rather than vote space. I don't know if we're going to end up doing a vote every week, so maybe it makes sense to make this number larger? Maybe make it 15 rounds (~1 month) or 30 rounds (~2 months)? Do you have any opinions here?